### PR TITLE
Bunch of fixes from several authors. 

### PR DESCRIPTION
**Pull request by cactusanctuary**
Update to use OpenFL imports instead of Flash.
Fix for Flash and HTML5 of TileGroup child visibility.
Fixes for Flash of TileSprite child visibility, offset,
Fixes for other targets of offset, pivot, rotation, texture scale
Also added a trace when trying to add a tile which doesn't exist in the tilesheet, so the developer can see which one is missing.

Authors: @elsassph, Joshua Granick

### CODE REVIEW

File: haxelib/aze/display/TileLayer.hx [hunk 1]
```diff
@@ -90,7 +90,11 @@ class TileLayer extends TileGroup
 			if (!child.visible) continue;
 			#end
 			
+			#if (flash||js)
```
Comment (elsassph) [inline haxelib/aze/display/TileLayer.hx]: Apparently this causes problems with neko, that's why it was simplified to a systematic use of Std.is.

File: haxelib/aze/display/TileLayer.hx [hunk 2]
```diff
@@ -177,14 +191,14 @@ class TileBase
 	public var animated:Bool;
 	public var visible:Bool;
 
-	function new(layer:TileLayer)
+	public function new(layer:TileLayer)
```
Comment (elsassph) [inline haxelib/aze/display/TileLayer.hx]: Why changing the methods to `public`?

File: haxelib/aze/display/TilesheetEx.hx [hunk 3]
```diff
@@ -80,6 +77,9 @@ class TilesheetEx extends Tilesheet
 				indices.push(i);
 		}
 		anims.set(name, indices);
+		if (indices.length == 0) {
```
Comment (elsassph) [inline haxelib/aze/display/TilesheetEx.hx]: Maybe put between `#if debug`

**Comment by elsassph** [comment]
There are lots of changes in the computations - does the sample still render correctly for all targets?

**Comment by elsassph** [comment]
Also would be useful to specify the version of lime/openfl to test against.

File: haxelib/aze/display/TileLayer.hx (hunk 1 above)
Comment (elsassph) [inline haxelib/aze/display/TileLayer.hx]: Should be `#if (flash || js || neko)`

**Comment by elsassph** [comment]
Alright, I updated everything to latest version and: flash/js are ok, but neko and C++ rotations are wrong. Run the test project to appreciate :)

**Comment by cactusanctuary** [comment]
Fixed rotations for neko and C++ targets, now these targets all look nice and consistent when testing with the test project: flash, js, mac neko, mac C++. 

Regarding the public access for DrawList constructor and functions, the compiler complains if they are private. Previously the public access was only #if haxe3, #else the DrawList implemented Public – extending Public allowed TileLayer to access DrawList's constructor and functions. Without it, each function needs to be made public.

**Pull request merged**